plugins/modules/scaleway_function.py

```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Scaleway Serverless function management module
+#
+# Copyright (c) 2022, Guillaume MARTINEZ <lunik@tiwabbit.fr>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: scaleway_function
+short_description: Scaleway Function management
+version_added: 6.0.0
+author: Guillaume MARTINEZ (@Lunik)
+description:
+  - This module manages function on Scaleway account.
+extends_documentation_fragment:
+  - community.general.scaleway
+  - community.general.scaleway_waitable_resource
+requirements:
+  - passlib[argon2] >= 1.7.4
+
+options:
+  state:
+    type: str
+    description:
+      - Indicate desired state of the function.
+    default: present
+    choices:
+      - present
+      - absent
+
+  namespace_id:
+    type: str
+    description:
+      - Function namespace identifier.
+    required: true
+
+  region:
+    type: str
+    description:
+      - Scaleway region to use (for example C(fr-par)).
+    required: true
+    choices:
+      - fr-par
+      - nl-ams
+      - pl-waw
+
+  name:
+    type: str
+    description:
+      - Name of the function.
+    required: true
+
+  description:
+    description:
+      - Description of the function.
+    type: str
+    default: ''
+
+  min_scale:
+    description:
+      - Minimum number of replicas for the function.
+    type: int
+
+  max_scale:
+    description:
+      - Maximum number of replicas for the function.
+    type: int
+
+  environment_variables:
+    description:
+      - Environment variables of the function.
+      - Injected in function at runtime.
+    type: dict
+    default: {}
+
+  secret_environment_variables:
+    description:
+      - Secret environment variables of the function.
+      - Updating thoses values will not output a C(changed) state in Ansible.
+      - Injected in function at runtime.
+    type: dict
+    default: {}
+
+  runtime:
+    description:
+      - Runtime of the function
+      - See U(https://www.scaleway.com/en/docs/compute/functions/reference-content/functions-lifecycle/) for all available runtimes
+    type: str
+    required: true
+
+  memory_limit:
+    description:
+      - Resources define performance characteristics of your function.
+      - They are allocated to your function at runtime.
+    type: int
+
+  function_timeout:
+    description:
+      - The length of time your handler can spend processing a request before being stopped.
+    type: str
+
+  handler:
+    description:
+      - The module-name.export value in your function.
```

Comment:
Not sure whether it fits:
```suggestion
      - The C(module-name.export) value in your function.
```